carRental/src/main/java/com/amigoscode/chohort2/carRental/spcefication/Util.java

```diff
@@ -0,0 +1,183 @@
+package com.amigoscode.chohort2.carRental.spcefication;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Expression;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import jakarta.persistence.metamodel.SingularAttribute;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+
+class Util {
+
+    private static final Logger log = LoggerFactory.getLogger(Util.class);
+
+
+    static <T> Predicate equal(Root<T> root, CriteriaBuilder cb,
+                               SingularAttribute<? super T, String> attributeName, String input) {
+        if (StringUtils.isNotBlank(input)) {
+            return cb.equal(root.get(attributeName), input.trim());
+        }
+        return null;
+    }
+
```

Comment:
I reckon, I'm gonna make use of the methods in Search implementation, aren't I ?!

Reply:
yes you will use more than one in fact

Reply:
Yeah that piece of code is the scariest one. :D